### ISSUE TITLE
define template location via configuration variable

### ISSUE THREAD

**Issue by fwoerister** [inveniosoftware/invenio-app-rdm#1053]
## Problem Description
We currently have the requirement to modify following templates in our custom theme:
- `invenio_app_rdm/help/search.en.html`
- `invenio_app_rdm/records/detail.html`
- `invenio_app_rdm/records/search.html`

Unfortunately we couldn't figure out an easy way (without messing around with the order of the registered blueprints) to implement this request.

The main problem is, that the according templates are hard-coded in the corresponding view functions (e.g. [in the view function of record_detail](https://github.com/inveniosoftware/invenio-app-rdm/blob/0010667e17d5a8470b84362bcf5f3835f7713041/invenio_app_rdm/records_ui/views/records.py#L74-L83)).

## Suggested Solution
I would suggest to specify this templates via config variables, similar as it is done for the frontpage (like [here](https://github.com/inveniosoftware/invenio-theme/blob/0fffb1d9af80a0216e478fced09c92891ce3d181/invenio_theme/views.py#L21-L26)).

## Alternative Approach
As already mentioned, there is a way to customize templates by making sure that the blueprint of our custom theme extension is placed before the blueprints of invenio-app-rdm. Flask is sequentially searching for a fitting templates, therefore a template in our custom theme would be selected before the invenio-app-rdm templates are checked. We currently have [a solution like that in place](https://gitlab.tuwien.ac.at/fairdata/invenio-theme-tuw/-/blob/master/invenio_theme_tuw/ext.py#L27-50) (but in my opinion this solution is too messy and should be avoided).

**Comment by fenekku** [comment]
What about the steps detailed here: https://inveniordm-qa.docs.cern.ch/customize/styling-templates/#change-the-record-landing-page ? Does that work for you? InvenioRDM should be supporting prioritized template loading by placing templates in a folder that jinja checks first.

If that doesn't work it's a bug to fix :)

**Comment by fwoerister** [comment]
Thanks for you reply :) Yes we already considered that (and it would work). But we are looking for a solution, where we can bundle all our customizations (templates, less and js files, etc.) in **one** extension.

But after taking a closer look on how the template lookup works, i think defining a new APP_THEME should do the trick.

**Comment by fenekku** [comment]
Yeah... I wanted this too originally (https://github.com/inveniosoftware/invenio-app-rdm/pull/638), but it wasn't kept I believe because we wanted to keep configuration low / revisit overall approach to configuration. :shrug: 

The `APP_THEME` was something we created to manage the gradual move from bootstrap to semantic-ui in invenio modules. I highly suspect that creating another `APP_THEME` won't quite work/will need a lot of dupliation because the key e.g. `semantic-ui` is then used in the webpack.py of many invenio-modules (you would have to somehow be able to replicate it...).

We can keep the ticket open to raise this up again (it probably won't see traction soon though).

**Comment by lnielsen** [comment]
The [``APP_THEME``](https://invenio-app.readthedocs.io/en/latest/configuration.html#invenio_app.config.APP_THEME) indeed does the job.

You would do something like:

```
APP_THEME = ['my-overlay', 'semantic-ui']
```

From the above example, templates and assets with the my-overlay prefix will be looked up first, and if not found the semantic-ui prefix will be used. If none of the lookups are successful, a non-prefixed lookup is done.

I'm moving this issue to the docs repository, since I think we just need to properly document how to do it. 

I'd suggest:
- Add a new guide: "Develop > Topic guides > Creating a theme" as I think it requires some development experience to do the job.
